src/main.cpp

```diff
@@ -881,26 +881,36 @@ bool CheckTransaction(const CTransaction& tx, CValidationState &state)
     // Ensure that pour values are well-formed
     BOOST_FOREACH(const CPourTx& pour, tx.vpour)
     {
-        if (pour.vpub_old < 0)
+        if (pour.vpub_old < 0) {
             return state.DoS(100, error("CheckTransaction(): pour.vpub_old negative"),
                              REJECT_INVALID, "bad-txns-vpub_old-negative");
+        }
 
-        if (pour.vpub_new < 0)
+        if (pour.vpub_new < 0) {
             return state.DoS(100, error("CheckTransaction(): pour.vpub_new negative"),
                              REJECT_INVALID, "bad-txns-vpub_new-negative");
+        }
 
-        if (pour.vpub_old > MAX_MONEY)
+        if (pour.vpub_old > MAX_MONEY) {
             return state.DoS(100, error("CheckTransaction(): pour.vpub_old too high"),
                              REJECT_INVALID, "bad-txns-vpub_old-toolarge");
+        }
 
-        if (pour.vpub_new > MAX_MONEY)
+        if (pour.vpub_new > MAX_MONEY) {
             return state.DoS(100, error("CheckTransaction(): pour.vpub_new too high"),
                              REJECT_INVALID, "bad-txns-vpub_new-toolarge");
+        }
+
+        if (pour.vpub_new =! 0 && pour.vpub_old != 0) {
```

Comment:
Wow! (It may have given a warning, but the mass of warnings the build already spits out has trained me to ignore them.)